### PR TITLE
Add Nostr SDK

### PR DESCRIPTION
**Pull request by mplorentz**
This adds the [Nostr SDK](https://github.com/nostr-sdk/nostr-sdk-ios) as a dependency. The main reason we need it is for its native Swift implementation of NIP-44 encryption which we need for #1015. 

It has a lot of other functionality we may find useful in the future too, although type name collisions might get annoying. I considered only pulling in the files we need, but decided to go this route as it will be easier to pull things in from the SDK on a case by case basis. If a few months go by and we aren't using it then we could trim the fat and just copy needed files. I did a "Build with Timing Summary" in Xcode 15.3 to see how much this adds to our build time, but it seems to be broken 💩. Thanks for nothing Xcode.

In order to add it we needed to upgrade our version of secp256k1, so this `closes` #1018. Fortunately almost 100% of the changed lines are covered by our unit tests! 🎉

### CODE REVIEW

**Comment by martindsq** [comment]
@mplorentz you are not using it here yet, right?

**Comment by mplorentz** [comment]
@martindsq correct! This just imports it and gets the project building again.

**Review by martindsq — APPROVED** [review]

File: Package.swift
Comment (joshuatbrown) [inline Package.swift:1]: @mplorentz what led to this addition?

Comment (mplorentz) [reply Package.swift:1]: Oops, this is something that helps out VSCode, it enables go to definition and some other features. I didn't mean to commit it here, but now that it's merged I kind of like that it's there. Maybe I will add a comment explaining what it is in another PR.

Comment (mplorentz) [reply Package.swift:1]: https://github.com/planetary-social/nos/pull/1059

**Pull request merged**